gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/fusion/functions/CoGroupAssociateOldVerticesWithNewIds.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of Gradoop.
+ *
+ * Gradoop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Gradoop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Gradoop. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.gradoop.flink.model.impl.operators.fusion.functions;
+
+import com.google.common.collect.Lists;
+import org.apache.flink.api.common.functions.CoGroupFunction;
+import org.apache.flink.api.java.functions.FunctionAnnotation;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.util.Collector;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Vertex;
+
+import java.util.Collection;
+
+/**
+ * Associates the old vertices with the new vertex fused ids.
+ *
+ */
+@FunctionAnnotation.ForwardedFieldsFirst("f0->f0")
+@FunctionAnnotation.ReadFieldsSecond("f0")
+public class CoGroupAssociateOldVerticesWithNewIds implements
```

Comment:
<img width="721" alt="schermata 2017-06-26 alle 10 44 28" src="https://user-images.githubusercontent.com/1330934/27531349-85b5467a-5a5c-11e7-9294-4453e7616400.png">

I think it breaks it.